controllers/storagecluster/cephrbdmirrors.go

```diff
@@ -0,0 +1,158 @@
+				return nil
+			}
+
+			if existing.DeletionTimestamp != nil {
+				r.Log.Info("Unable to restore CephRbdMirror because it is marked for deletion.", "CephRbdMirror", klog.KRef(existing.Namespace, existing.Name))
+				return fmt.Errorf("failed to restore initialization object %s because it is marked for deletion", existing.Name)
+			}
+
+			r.Log.Info("Restoring original CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+			existing.ObjectMeta.OwnerReferences = cephRbdMirror.ObjectMeta.OwnerReferences
+			cephRbdMirror.ObjectMeta = existing.ObjectMeta
+			err = r.Client.Update(context.TODO(), cephRbdMirror)
+			if err != nil {
+				r.Log.Error(err, "Failed to update CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+				return err
+			}
+		case errors.IsNotFound(err):
+			r.Log.Info("Creating CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+			err = r.Client.Create(context.TODO(), cephRbdMirror)
+			if err != nil {
+				r.Log.Error(err, "Failed to create CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// ensureDeleted deletes the CephRbdMirrors owned by the StorageCluster
+func (obj *ocsCephRbdMirrors) ensureDeleted(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) error {
+	foundCephRbdMirror := &cephv1.CephRBDMirror{}
+	cephRbdMirrors, err := r.newCephRbdMirrorInstances(sc)
+	if err != nil {
+		return err
+	}
+
+	for _, cephRbdMirror := range cephRbdMirrors {
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: cephRbdMirror.Name, Namespace: sc.Namespace}, foundCephRbdMirror)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				r.Log.Info("Uninstall: CephRBDMirror not found.", "CephRBDMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+				continue
+			}
+			return fmt.Errorf("uninstall: unable to retrieve CephRbdMirror %v: %v", cephRbdMirror.Name, err)
+		}
+
+		err = r.deleteCephRbdMirrorInstance(cephRbdMirror, foundCephRbdMirror)
+		if err != nil {
+			r.Log.Error(err, "Uninstall: Failed to delete CephRbdMirror.", "CephRbdMirror", klog.KRef(foundCephRbdMirror.Namespace, foundCephRbdMirror.Name))
+			return fmt.Errorf("Uninstall: Failed to delete CephRbdMirror %v: %v", foundCephRbdMirror.Name, err)
+		}
+
+		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: cephRbdMirror.Name, Namespace: sc.Namespace}, foundCephRbdMirror)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				r.Log.Info("Uninstall: CephRbdMirror is deleted.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+				continue
+			}
+		}
+		r.Log.Error(err, "Uninstall: Waiting for CephRbdMirror to be deleted.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+		return fmt.Errorf("uninstall: Waiting for CephRbdMirror %v to be deleted", cephRbdMirror.Name)
```

Comment:
This part is executed when there is **no** error. Question is, if there is no error, why are we making sure once again (is it a norm or practice, that we have to do a double check) the resources doesn't exist.

Reply:
It is not needed.
Try to delete it, if it fails retry deletion. That's it.